network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py

```diff
@@ -87,6 +87,20 @@ def filter_entries(self, entries, context):
     def filter_entries_for_category(self, entries, context):
         category = self.filtered.get('category')
 
+        # The following "if" statements update page share metadata when filtered by category.
+        #
+        # Update page search/share metadata to be the category's description.
+        # If not set, default to category's "intro" text.
+        # If "intro" is not set, use the foundation's default meta description.
+        if category.share_description:
+            setattr(self, 'search_description', category.share_description)
+        elif not category.share_description and category.intro:
```

Comment:
we already know that `category.share_description` is falsey here, because we just tested for it =)

```suggestion
        elif category.intro:
```